plugins/inputs/github/README.md

```diff
@@ -0,0 +1,48 @@
+# GitHub Input Plugin
+
+The [GitHub](https://www.github.com) input plugin gathers statistics from GitHub repositories.
+
+### Configuration:
+
+```toml
+[[inputs.github]]
+  ## List of repositories to monitor
+  ## ex: repositories = ["influxdata/telegraf"]
+  # repositories = []
+
+  ## Optional: Unauthenticated requests are limited to 60 per hour.
+  # access_token = ""
+
+  ## Optional: Default 5s.
+  # http_timeout = "5s"
+```
+
+### Metrics:
+
+- github_repository
+  - tags:
+    - `full_name` - The full name of the repository, including owner/organization
+    - `name` - The repository name
+    - `owner` - The owner of the repository
+    - `language` - The primary language of the repository
+    - `license` - The license set for the repository
+  - fields:
+    - `stars` (int)
+    - `forks` (int)
+    - `open_issues` (int)
+    - `size` (int)
+
+* github_rate_limit
+  - tags:
+    - `access_token` - An obfusticated reference to the configured access token or "Unauthenticated"
+  - fields:
+    - `limit` - How many requests you are limited to (per hour)
+    - `remaining` - How many requests you have remaining (per hour)
+    - `blocks` - How many requests have been blocked due to rate limit
+
+### Example Output:
+
+```
+github,full_name=influxdata/telegraf,name=telegraf,owner=influxdata,language=Go,license=MIT\ License stars=6401i,forks=2421i,open_issues=722i,size=22611i 1552651811000000000
+github_rate_limit,access_token=Unauthenticated remaining=59i,limit=60i,blocks=0i 1552653551000000000
```

Comment:
Update for `internal` metric style.

I would use the repo name and owner as tags, this will give you number of blocks per repo and sidesteps the access_token encoding.

Reply:
I've updated the internal metric example.

I don't think we should use the repo name and owner as tags, as the block is associated with the access token, the repo is irrelevant.

Reply:
I think there are a couple arguments for repo:
- some users may be uncomfortable with access_token encoding, even though it is limited.
- there isn't always an access_token, requiring a special value.  this isn't necessarily bad but it is a code smell to me.
- it may be interesting to know which repository was blocked during a time period

I'll leave the final decision for you, I'm okay with how it is as well.

Reply:
I can see both sides, but as the block is directly linked to the access token; I'm keen to leave it as is and maybe we can collect some user feedback.

Happy for this to be merged as is. Thanks for your feedback and help during this, appreciated.